### PR TITLE
Relocate the Knative mirror job

### PR DESCRIPTION
**Pull request by matzew**
@smarterclayton wants the knative image mirror job to be in a different spot. For now, I've moved them into their own folder.

Happy to change the folder structure/location, if changes are needed or desired

/cc @markusthoemmes @vdemeester @pmorie

### CODE REVIEW

File: cluster/ci/config/mirroring/knative/OWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+approvers:
+- smarterclayton
```
Comment (matzew) [inline cluster/ci/config/mirroring/knative/OWNERS]: @stevekuznetsov @smarterclayton feel free to adjust/add/remove you :)

**Comment by matzew** [comment]
@pmorie @smarterclayton  Does this seem reasonable to relocated the knative image mirror stuff to a different spot ?

**Comment by matzew** [comment]
> Update the `Makefile` to make sure you get auto-updated on merge

@stevekuznetsov thanks for looking. Let me please ask.  Yoy mean the `image-mirror-files` section the in the `Makefile`, which than also means to create a similar file like `hack/mirror-file` for the knative images?

**Comment by stevekuznetsov** [comment]
I'm not opinionated as to how you do it but as some sub-target of `make postsubmit-update` please have an auto-update for the YAML here. The `config-updater` plugin is used to push the actual mirror configs up (sorry, that's confusing) -- look a `cluster/ci/config/prow/plugins.yaml`

**Comment by matzew** [comment]
@stevekuznetsov thanks for the reply. I am not 100% sure what exactly you mean.

I did assume that the _current_ `image-mirror.yaml` has this already, but apparently not?

So I should add an `arg` for `postsubmit-update` in the jobs, for the different mirror/job files?

**Comment by matzew** [comment]
@petr-muller did help me a lot on the slack channel w/ adding the hook to the `prow` plugin.

thanks a million, Petr :tada:

**Comment by stevekuznetsov** [comment]
@matzew can you squash your commits?

**Comment by matzew** [comment]
@stevekuznetsov done!

**Comment by pmorie** [comment]
@smarterclayton is there anything we need to do to get an /approve from you?

File: cluster/ci/jobs/image-mirror.yaml [hunk 2]
```diff
@@ -278,5 +278,52 @@ items:
...
+            volumes:
+            - name: push
+              secret:
+                secretName: registry-push-credentials-quay.io-openshift-knative
+            - name: config
+              configMap:
+                name: image-mirror
+            containers:
+            - name: "mirror"
+              image: "openshift/origin-cli:v3.10"
```
Comment (smarterclayton) [inline cluster/ci/jobs/image-mirror.yaml]: change to v4.0

Comment (matzew) [reply cluster/ci/jobs/image-mirror.yaml]: @smarterclayton   done!

**Pull request merged**